api/rest/restcore/users_rest.php

```diff
@@ -81,6 +91,131 @@ function rest_user_create( \Slim\Http\Request $p_request, \Slim\Http\Response $p
+ */
+function rest_user_create_token_for_current_user( \Slim\Http\Request $p_request, \Slim\Http\Response $p_response, array $p_args ) {
+	$t_user_id = auth_get_current_user_id();
+	return execute_create_token_command( $p_request, $p_response, $t_user_id );
+}
+
+/**
+ * Helper method for creation of user tokens
+ *
+ * @param \Slim\Http\Request $p_request   The request.
+ * @param \Slim\Http\Response $p_response The response.
+ * @param integer $p_user_id The id of the user to create token for.
+ *
+ * @return \Slim\Http\Response The augmented response.
+ */
+function execute_create_token_command( \Slim\Http\Request $p_request, \Slim\Http\Response $p_response, $p_user_id ) {
+	// if body is empty or {} it will fail, this is acceptable for this API.
+	$t_payload = $p_request->getParsedBody();
+	if( !$t_payload ) {
+		$t_payload = array();
+	}
+
+	$t_data = array(
+		'query' => array(
+			'user_id' => (int)$p_user_id
+		),
+		'payload' => $t_payload
+	);
+
+	$t_command = new UserTokenCreateCommand( $t_data );
+	$t_result = $t_command->execute();
+
+	return $p_response->withStatus( HTTP_STATUS_CREATED, "User token created" )->
+		withJson( $t_result );
+}
+
+/**
+ * A method that deletes a user token for another user.
+ *
+ * @param \Slim\Http\Request $p_request   The request.
+ * @param \Slim\Http\Response $p_response The response.
+ * @param array $p_args Arguments
+ *
+ * @return \Slim\Http\Response The augmented response.
+ *
+ * @noinspection PhpUnusedParameterInspection
+ */
+function rest_user_delete_token( \Slim\Http\Request $p_request, \Slim\Http\Response $p_response, array $p_args ) {
+	$t_token_id = $p_args['id'];
+	$t_user_id = $p_args['user_id'];
+	return execute_delete_token_command( $p_request, $p_response, $t_user_id, $t_token_id );
+}
+
+/**
+ * A method that deletes a user token for current user.
+ *
+ * @param \Slim\Http\Request $p_request   The request.
+ * @param \Slim\Http\Response $p_response The response.
+ * @param array $p_args Arguments
+ *
+ * @return \Slim\Http\Response The augmented response.
+ *
+ * @noinspection PhpUnusedParameterInspection
+ */
+function rest_user_delete_token_for_current_user( \Slim\Http\Request $p_request, \Slim\Http\Response $p_response, array $p_args ) {
+	$t_token_id = $p_args['id'];
+	$t_user_id = auth_get_current_user_id();
+	return execute_delete_token_command( $p_request, $p_response, $t_user_id, $t_token_id );
+}
+
+/**
+ * Helper method for creation of user tokens
+ *
+ * @param \Slim\Http\Request $p_request   The request.
+ * @param \Slim\Http\Response $p_response The response.
+ * @param integer $p_user_id The id of the user to create token for.
+ *
+ * @return \Slim\Http\Response The augmented response.
+ */
+function execute_delete_token_command( \Slim\Http\Request $p_request, \Slim\Http\Response $p_response, $p_user_id, $p_token_id ) {
+	// if body is empty or {} it will fail, this is acceptable for this API.
+	$t_payload = $p_request->getParsedBody();
```

Comment:
`$t_payload` is not used in this function

Reply:
Fixed.